### PR TITLE
Website

### PR DESCRIPTION
**Pull request by aquaductape**
Website for solid-primitives

Still need to figure out on how it should be deployed since demo pages uses /solid-primitives/ base for github pages, and in my opinion this site should use that base and the demo pages be moves elsewhere, like in a subdirectory like `demo`, `/solid-primitives/demo/activeElement...` (pretty sure it's possible for gh-pages).

As far as order of build script for successfully running site
1. `pnpm build` (necessary to for esbuild script to read export sizes from packages that is displayed in the website)
2. `pnpm update-site`
3. `cd site`
4. `pnpm install`
5. `pnpm dev` or build -> start

### CODE REVIEW

**Comment by atk** [comment]
We can rename the demo page index for now and try to integrate it into the site later.

**Comment by aquaductape** [comment]
Is my prettier broken, I notice there's a ton of files such as https://github.com/solidjs-community/solid-primitives/pull/338/files#diff-df2dff4c463317a70a8590772e15bf2905cef2d70a2f0b2b209fc09ed8898199 where my PR replaced tabs (or maybe 4 spaces) with 2 spaces

**Comment by atk** [comment]
I think 2 spaces is the preferred formatting, but you should only run prettier on new files; we should probably have a separate cleaup PR.

**Comment by thetarnav** [comment]
make sure you have prettier as the default formatter
otherwise, you're going to fight with our format gh action 😅

image

**Comment by thetarnav** [comment]
There are some additional changes to the source of a few packages. This looks like a merge artifact.

<details>

image

image

image

image

image
</details>

**Comment by thetarnav** [comment]
And about the deployment. Could we have this website and the playground demos deployed to netlify / vercel instead of gh-pages so that we get deployment previews on PRs?

**Comment by aquaductape** [comment]
> I think 2 spaces is the preferred formatting, but you should only run prettier on new files; we should probably have a separate cleaup PR.

To make it easier I just removed the changes of the html files from dev, some html files from main branch aren't properly prettiefied

**Comment by aquaductape** [comment]
> There are some additional changes to the source of a few packages. This looks like a merge artifact.

@thetarnav fixed 7828991341887764e3721eab4c0b88430826adee

**Comment by aquaductape** [comment]
Not sure how to fix regex to satisfy CodeQL errors

**Comment by aquaductape** [comment]
Not sure how to solve these two CodeQL errors.

Screenshot 2023-03-02 at 7 42 07 PM
The tailwind one is absolutely ridiculous, wrote a simple plugin class, (to make a dashed border that's 100% visually consistent css border dash has different spacing across browsers)  that replaces `#` to url encoded value `%23`.

**Comment by atk** [comment]
You're probably supposed to use encodeURIComponent instead of a hand crafted replace.

**Comment by thetarnav** [comment]
wow you've defeated the codeql alerts 👏👏🤯
I give up as soon as I see them

**Comment by aquaductape** [comment]
> wow you've defeated all the codeql alerts 👏👏🤯 I give up as soon as I see them

discovering catastrophic backtracking made me depressed, regex is already hard as it is

**Comment by aquaductape** [comment]
> wow you've defeated all the codeql alerts 👏👏🤯 I give up as soon as I see them

The "Quickly Matching a Complete HTML File" section in https://www.regular-expressions.info/catastrophic.html helped me, but then Javascript doesn't support atomic groups, so this one helped me https://blog.stevenlevithan.com/archives/mimic-atomic-groups

File: scripts/update-site/build-pages.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+      return `${p1}<NoHydration>${content}</NoHydration>${p3}`;
+    })
+    // update code tag that contains primitives to have attribute
+    .replace(primitiveCodeElRegex, (_, p1, p2, p3) => {
+      if (!p2) return _;
+      return `<${p1} data-code-primitive-name="${p2.replace(
+        /\<|\>/g,
+        "",
+      )}" data-code-package-name="${name}">{"${p2}"}</${p3}>`;
```
[the remark this answers is not in the document]
Comment (aquaductape) [reply scripts/update-site/build-pages.ts:130]: phew it's a warning, this regex isn't sanitizing, so all good

**Pull request merged**